util/dockerfiles/Dockerfile

```diff
@@ -54,4 +55,9 @@ RUN sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen && \
 RUN git config --global user.email "noreply@example.com" && \
     git config --global user.name  "Chapel user"
 
-ENV PATH $PATH:$CHPL_HOME/bin/linux64-x86_64:$CHPL_HOME/util
+
+COPY entrypoint.sh .
+RUN chmod a+x /entrypoint.sh
+
+# setting path based on the underlying architecture
+ENTRYPOINT ["/entrypoint.sh"]
```

Comment:
I know very little about creating dockerfiles, why do we need a separate script to do this?

Reply:
I am not very proficient in docker as well.

From what I understand about using entrypoint is to run a set of commands inside the container as a main function and it is best practice to send a file.sh as parameter.

Reply:
To expand on this, the `ENTRYPOINT` defines what happens when someone `docker run`s the container. I would expect this to cause `docker run chapel/chapel` to drop you into a shell in a newly-created container, whereas without an `ENTRYPOINT` you'd need to do `docker run chapel/chapel /bin/bash` to get equivalent behavior (not to mention the env vars wouldn't get set). [Reference](https://docs.docker.com/engine/reference/builder/#entrypoint)